components/server/src/projects/projects-service.ts

```diff
@@ -554,21 +555,38 @@ export class ProjectsService {
     ): Promise<WebhookEvent | undefined> {
         const context = (await this.contextParser.handle(ctx, user, project.cloneUrl)) as CommitContext;
 
-        const events = await this.webhookEventDb.findByCloneUrl(project.cloneUrl, 1);
+        const events = await this.webhookEventDb.findByCloneUrl(project.cloneUrl, 50);
```

Comment:
`prompt("Why especially 50? Could we have chosen a different number as well? What's the tradeoff?") > comment` :orange_heart:

Reply:
Thanks for the comment reminder, will add and push 🙏

Reply:
Pushed as [`c071537` (#20240)](https://github.com/gitpod-io/gitpod/pull/20240/commits/c071537a9a9c3936358eccdee57aff26ce8f4cd4)